Sources/Rendering/Core/ImageCPRMapper/index.js

```diff
@@ -0,0 +1,350 @@
+import CoincidentTopologyHelper from 'vtk.js/Sources/Rendering/Core/Mapper/CoincidentTopologyHelper';
+import vtkAbstractImageMapper from 'vtk.js/Sources/Rendering/Core/AbstractImageMapper';
+import vtkPolyLine from 'vtk.js/Sources/Common/DataModel/PolyLine';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import macro from 'vtk.js/Sources/macros';
+import { mat3, mat4, quat, vec3 } from 'gl-matrix';
+
+const { vtkErrorMacro } = macro;
+
+const { staticOffsetAPI, otherStaticMethods } = CoincidentTopologyHelper;
+
+// ----------------------------------------------------------------------------
+// vtkImageCPRMapper methods
+// ----------------------------------------------------------------------------
+
+function vtkImageCPRMapper(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkImageCPRMapper');
+
+  /**
+   * Public methods
+   */
+  publicAPI.getBounds = () => {
+    const imageWidth = publicAPI.getWidth();
+    const imageHeight = publicAPI.getHeight();
+    return [0, imageWidth, 0, imageHeight, 0, 0];
+  };
+
+  publicAPI.getOrientationDataArray = () => {
+    const pointData = publicAPI.getInputData(1)?.getPointData();
+    if (!pointData) {
+      return null;
+    }
+    if (model.orientationArrayName !== null) {
+      return pointData.getArrayByName(model.orientationArrayName) || null;
+    }
+    return (
+      pointData.getArrayByName('Orientation') ||
+      pointData.getArrayByName('Direction') ||
+      pointData.getVectors() ||
+      pointData.getTensors() ||
+      pointData.getNormals() ||
+      null
+    );
+  };
+
+  publicAPI.getOrientedCenterline = () => {
+    const inputPolydata = publicAPI.getInputData(1);
+    if (inputPolydata) {
+      // Get dependencies of centerline
+      const orientationDataArray = publicAPI.getOrientationDataArray();
+      const linesDataArray = inputPolydata.getLines();
+      const pointsDataArray = inputPolydata.getPoints();
+
+      if (!model.useUniformOrientation && !orientationDataArray) {
+        vtkErrorMacro(
+          'Failed to create oriented centerline from polydata: no orientation'
+        );
+        model._orientedCenterline = vtkPolyLine.newInstance();
```

Comment:
It seems better to create a polyline only once in constructor, and reset it here.